JLio.Functions/FilterBySchema.cs

```diff
@@ -0,0 +1,104 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using JLio.Core;
+using JLio.Core.Contracts;
+using JLio.Core.Extensions;
+using JLio.Core.Models;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+namespace JLio.Functions
+{
+    public class FilterBySchema : FunctionBase
+    {
+        public FilterBySchema()
+        {
+        }
+
+        public FilterBySchema(JSchema schema)
+        {
+            Arguments.Add(new FunctionSupportedValue(new FixedValue(schema)));
+        }
+
+        public FilterBySchema(string path)
+        {
+            Arguments.Add(new FunctionSupportedValue(new FixedValue(JSchema.Parse($"\"{path}\""))));
+        }
+        
+        public override JLioFunctionResult Execute(JToken currentToken, JToken dataContext, IExecutionContext context)
+        {
+            if (Arguments.Count == 0 || Arguments.Count > 1)
+            {
+                context.LogError(CoreConstants.FunctionExecution,
+                    $"failed: {FunctionName} requires one argument: path of the schema or the schema itself");
+                return JLioFunctionResult.Failed(currentToken);
+            }
+            
+            var values = GetArguments(Arguments, currentToken, dataContext, context);
+
+            var pathsToFilter = values[0].ToObject<JSchema>().GetPaths();
+            
+            var inputObjectPaths = GetInputPaths(currentToken);
+
+            var pathsInSchema = pathsToFilter.Select(s => s.Path).ToList();
+            
+            foreach (var pathToRemove in GetPathsToRemove(inputObjectPaths, pathsInSchema))
+            {
+                var token = currentToken.SelectToken(pathToRemove);
+                if (token != null)
+                {
+                    RemoveItems(currentToken, pathToRemove, context);
+                }
+            }
+            return new JLioFunctionResult(true, currentToken);
+        }
+
+        private List<string> GetPathsToRemove(List<string> objectPaths, List<string> pathsToFilter)
+        {
+            return  objectPaths.Except(pathsToFilter).ToList();
+        }
+
+        private List<string> GetInputPaths(JToken input)
+        {
+            var paths = new List<string>();
+            var tokens = JsonMethods.GetAllElements(input);
+            foreach (var jsonToken in tokens)
+            {
+                if (jsonToken.Path == string.Empty)
+                {
+                    continue;
+                }
+                paths.Add(GetPath(jsonToken.Path));
+            }
+
+            return paths;
+        }
+
+        private string GetPath(string path)
+        {
+            var result = $"$.{path}";
+            if (result.Contains("["))
```

Comment:
this is not fully correct.
json properties allow "[" in their name.
for example you can have this json

```json
{
  "something[with-brackets]": "my_value"
}
```
In this case a valid path is:
```$["something[with-brackets]"]```
You need to make sure that the brackets are not part of the name of the property.
Also, you can't rely on getting rid of the  contents of what is inside the brackets. The brackets don't necessarily mean array / query.

For example the path ```$.a.b.c``` can also be written equivalently as ```$["a"]["b"]["c"]```
This is actually how you can do a path in case your json property has spaces in it, because you can't use the simple "dot"(.) notation

Reply:
be aware that the notion is with a single quote not a double

$["a"]["b"]["c"]

let's make a full collection of special cases in the path notations. Create a new item on the board and implement those special cases in the unittest for the different components that uses them. 
For now the selection can be to the normal ones , and we address this as an know issue

Reply:
added logic for the path suport in Feature/expand support for jsonpath notations